### DISCUSSION TITLE
Configuring the max data chunk size returned in QueryResult

### DISCUSSION THREAD

**Discussion by chu-k** [duckdb/duckdb#13502] [Q&A] [answered]
The C API suggests that the maximum number of rows (VECTOR_SIZE) in a DataChunk [can be configured](https://duckdb.org/docs/api/c/data_chunk). I can't seem to find the API method do to so, or a corresponding configuration option.

**Comment by Tishj** [comment, marked as answer]
That is a compile-time option, not changable at runtime

**Reply by chu-k** [reply]
[Found it](https://github.com/duckdb/duckdb/blob/862852fa395b99735e5713cb55d0cea1d9320659/CMakeLists.txt#L504), thank you for the pointer. 

It seems like this only affects the [default capacity.](https://github.com/duckdb/duckdb/blob/862852fa395b99735e5713cb55d0cea1d9320659/src/common/types/data_chunk.cpp#L22) i.e., I should still be able to exceed this when explicitly constructing a DataChunk with Initialize for use with AppendDataChunk (C++ API) is that correct?
